sqlmesh/core/context_diff.py

```diff
@@ -83,26 +83,39 @@ def create(
         if env is None:
             env = state_reader.get_environment(create_from.lower())
             is_new_environment = True
-            previously_promoted_model_names = set()
+            previously_promoted_snapshot_ids = set()
         else:
             is_new_environment = False
-            previously_promoted_model_names = {s.name for s in env.promoted_snapshots}
-
-        existing_info = {info.name: info for info in (env.snapshots if env else [])}
-        existing_nodes = set(existing_info)
-        current_nodes = set(snapshots)
-        removed = existing_nodes - current_nodes
-        added = current_nodes - existing_nodes
-        modified_info = {
-            name: existing_info[name]
-            for name, snapshot in snapshots.items()
-            if name not in added and snapshot.fingerprint != existing_info[name].fingerprint
-        }
+            previously_promoted_snapshot_ids = {s.snapshot_id for s in env.promoted_snapshots}
 
+        environment_snapshot_infos = env.snapshots if env else []
+        remote_snapshot_name_to_info = {
+            snapshot_info.name: snapshot_info for snapshot_info in environment_snapshot_infos
```

Comment:
why are you using name and not fqn?

Reply:
has any logic changed in this whole section except that you're not using snapshot ids?

Reply:
Yes and also changed the approach a bit like building removed with the correct data structure from the start instead of creating it at the end.